python/lsst/meas/base/applyApCorr.py

```diff
@@ -0,0 +1,159 @@
+        )
+
+class ApplyApCorrConfig(lsst.pex.config.Config):
+    ignoreList = lsst.pex.config.ListField(
+        doc = "flux measurement algorithms in getApCorrNameSet() to ignore;" +
+            " if a name is listed that does not appear in getApCorrNameSet() then a warning is logged",
+        dtype = str,
+        optional = False,
+        default = (),
+    )
+    doFlagApCorrFailures = lsst.pex.config.Field(
+        doc = "set the general failure flag for a flux when it cannot be aperture-corrected?",
+        dtype = bool,
+        default = True,
+    )
+
+
+class ApplyApCorrTask(lsst.pipe.base.Task):
+    """!Apply aperture corrections
+    """
+    ConfigClass = ApplyApCorrConfig
+    _DefaultName = "applyApCorr"
+
+    def __init__(self, schema, **kwds):
+        lsst.pipe.base.Task.__init__(self, **kwds)
+
+        self.apCorrInfoDict = dict()
+        apCorrNameSet = getApCorrNameSet()
+        ignoreSet = set(self.config.ignoreList)
+        missingNameSet = ignoreSet - set(apCorrNameSet)
+        if missingNameSet:
+            self.log.warn("Fields in ignoreList that are not in fluxCorrectList: %s" %
+                (sorted(list(missingNameSet)),))
+        for algName in apCorrNameSet - ignoreSet:
+            if algName + "_flux" in schema:
+                self.apCorrInfoDict[algName] = ApCorrInfo(schema=schema, algName=algName)
+            else:
+                # if a field in the registry is missing from the schema, silently ignore it.
+                pass
+
+    def run(self, catalog, apCorrMap):
+        """Apply aperture corrections to a catalog of sources
+
+        @param[in,out] catalog  catalog of sources
+        @param[in] apCorrMap  aperture correction map (an lsst.afw.image.ApCorrMap)
+        """
+
+        for apCorrInfo in self.apCorrInfoDict.itervalues():
+            try:
+                apCorrModel = apCorrMap.get(apCorrInfo.fluxName)
+                apCorrSigmaModel = apCorrMap.get(apCorrInfo.fluxSigmaName)
+            except Exception:
+                self.log.warn("Could not find %s in apCorrMap" % (apCorrInfo.algName,))
+                for source in catalog:
+                    source.set(apCorrInfo.apCorrFlagKey, True)
+                continue
+
+            for source in catalog:
+                center = source.getCentroid()
+                # say we've failed when we start; we'll unset these flags when we succeed
+                source.set(apCorrInfo.apCorrFlagKey, True)
+                oldFluxFlagState = False
+                if self.config.doFlagApCorrFailures:
+                    oldFluxFlagState = source.get(apCorrInfo.fluxFlagKey)
+                    source.set(apCorrInfo.fluxFlagKey, True)
+
+                apCorr = 1.0
+                apCorrErr = 0.0
+                try:
+                    apCorr = apCorrModel.evaluate(center)
+                    apCorrErr = apCorrSigmaModel.evaluate(center)
+                except Exception:
+                    continue
```

Comment:
Would like a tighter exception specifier here (I realize that may be a problem you inherited from the HSC side).